frontend-web/webclient/app/Grants/Editor.tsx

```diff
@@ -271,10 +276,17 @@ function stateReducer(state: EditorState, action: EditorAction): EditorState {
 
                     const existing = sectionForProvider.find(it => it.category.name === category.name);
                     if (existing) {
-                        existing.allocators.add(allocator.id);
+                        if (existing.allocators.values().find(grantGiver => grantGiver.grantGiverId == allocator.id && grantGiver.grantGiverTitle === allocator.title)) {
+                            //DO NOTHING
+                        } else {
+                            existing.allocators.add({grantGiverId: allocator.id, grantGiverTitle: allocator.title});
+                        }
                     } else {
-                        const allocators = new Set<string>();
-                        allocators.add(allocator.id);
+                        const allocators = new Set<GrantGiverIdAndTitle>();
+                        const exists = Array.from(allocators.entries()).find(grantgiver => grantgiver.grantGiverId === allocator.id && grantgiver.grantGiverTitle === allocator.title);
```

Comment:
Am I misunderstanding or are you creating a new empty set an then checking it for entries?